experimental/material_3_demo/lib/main.dart

```diff
@@ -0,0 +1,202 @@
+// Copyright 2021 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'color_palettes_screen.dart';
+import 'component_screen.dart';
+import 'elevation_screen.dart';
+import 'typography_screen.dart';
+
+void main() {
+  runApp(const Material3Demo());
+}
+
+class Material3Demo extends StatefulWidget {
+  const Material3Demo({super.key});
+
+  @override
+  State<Material3Demo> createState() => _Material3DemoState();
+}
+
+// NavigationRail shows if the screen width is greater or equal to
+// screenWidthThreshold; otherwise, NavigationBar is used for navigation.
+const double narrowScreenWidthThreshold = 450;
+
+const Color m3BaseColor = Color(0xff6750a4);
+const List<Color> colorOptions = [
+  m3BaseColor,
+  Colors.blue,
+  Colors.teal,
+  Colors.green,
+  Colors.yellow,
+  Colors.orange,
+  Colors.pink
+];
+const List<String> colorText = <String>[
+  "M3 Baseline",
+  "Blue",
+  "Teal",
+  "Green",
+  "Yellow",
+  "Orange",
+  "Pink",
+];
+
+class _Material3DemoState extends State<Material3Demo> {
+  bool useMaterial3 = true;
+  bool useLightMode = true;
+  int colorSelected = 0;
+  int screenIndex = 0;
```

Comment:
Might be a little more self-explanatory as an enum (modern Dart enums can have properties).